docs/docs/code/go/README.md

```diff
@@ -1,6 +1,452 @@
+	body, _ := ioutil.ReadAll(resp.Body)
+
+  // Unmarshal the JSON into a struct
+	var data map[string]interface{}
+	json.Unmarshal(body, &data)
+
+  // Expose the pokemon data downstream to others steps in the "pokemon" key from this step
+	pd.Export("pokemon", data)
+}
+```
+
+Now this `pokemon` data is accessible to downstream steps within `pd.Steps.code.pokemon`
+
+::: warning
+Not all data types can be stored in the `Steps` data shared between workflow steps.
+
+For the best experience, we recommend only [exporting structs that can be marshalled into JSON](https://go.dev/blog/json).
+
+[Read more details on step limitations here.](/workflows/steps/#limitations-on-step-exports)
+:::
+
+## Using environment variables
+
+You can leverage any [environment variables defined in your Pipedream account](/environment-variables/#environment-variables) in a Python step. This is useful for keeping your secrets out of code as well as keeping them flexible to swap API keys without having to update each step individually.
+
+To access them, use the `os` package.
+
+```go
+package main
+
+import (
+  "log"
+  "os"
+)
+
+func main() {
+  log.PrintLn(os.Getenv('TWITTER_API_KEY'))
+}
+```
+
+Or an even more useful example, using the stored environment variable to make an authenticated API request.
+
+### Using API key authentication
+
+If an particular service requires you to use an API key, you can pass it via the headers of the request.
+
+This proves your identity to the service so you can interact with it:
+
+```go
+package main
+
+import (
+  "os"
+  "bytes"
+  "encoding/json"
+  "io/ioutil"
+  "log"
+  "net/http"
+  "fmt"
+)
+
+func main() {
+  // Access the Twitter API key from the environment 
+  const apiKey := os.Getenv('TWITTER_API_KEY'))
+
+  // JSON encode our payload
+  payload, _ := json.Marshal(map[string]string{
+    "name":  "Bulbasaur",
+  })
+  payloadBuf:= bytes.NewBuffer(payload)
+
+  // Send the POST request
+  req, err := http.NewRequest("POST", "https://api.twitter.com/2/users/@pipedream/mentions'", payloadBuf)
+   
+  // Build the headers in order to authenticate properly
+  req.Header = http.Header{
+    "Content-Type": []string{"application/json"},
+    "Authorization": []string{fmt.Sprintf("Bearer %s", apiKey)},
+  }
+
+  client := http.Client{}
+  resp, err := client.Do(req)
+  
+  if err != nil {
+    log.Fatalln(err)
+  }
+  // Don't forget to close the request after the function is finished
+  defer resp.Body.Close()
+
+  // Read the response body
+  body, err := ioutil.ReadAll(resp.Body)
+  if err != nil {
+    log.Fatalln(err)
+  }
+  // Convert the body into a string
+  sb := string(body)
+  // Log the body to our Workflow Results
+  log.Println(sb)
+}
+```
+
+## Handling errors
+
+You may need to exit a workflow early. In a Go step, just a `return` from the `main` function.
+
+Optionally, you may want to return an `error` for logging.
+
+
+```go
+package main
```

Comment:
This code fails with other errors (i.e. it doesn't throw the error you included in code)